src/components/ChatMode/index.vue

```diff
@@ -30,23 +30,50 @@
                 class="v-content__wrap fill-height"
               >
                 <v-col
-                  class="my-1 bubble-container"
+                  v-for="({ node, componentType }, index) in allNodes"
+                  :class="{
+                    'my-1': true,
+                    'bubble-container': true,
+                    'grp-with-prev-msg': shouldGroupWithPrevMessage(
+                      node,
+                      (index - 1 >= 0) ? allNodes[index - 1].node : null,
+                      (index + 1 < allNodes.length) ? allNodes[index + 1].node : null,
+                    ),
+                    'grp-with-next-msg': shouldGroupWithNextMessage(
+                      node,
+                      (index - 1 >= 0) ? allNodes[index - 1].node : null,
+                      (index + 1 < allNodes.length) ? allNodes[index + 1].node : null,
+                    )
+                  }"
                   cols="12"
-                  v-for="{ node, componentType } in allNodes"
                   :data-side="fromUser(node) ? 'user' : 'bot'"
                   :key="node.id"
                 >
                   <v-row
                     dense
                     :class="{
-                      'pa-1': true,
+                      'px-1': true,
                       'flex-row-reverse': fromUser(node),
                     }"
                   >
-                    <v-col cols="auto" v-if="!fromUser(node)">
-                      <img height="40" width="40" :src="avatar" mr-1 />
+                    <v-col class="py-0" cols="auto" v-if="!fromUser(node)" style="height: 40px;">
+                      <img 
+                        v-if="!shouldGroupWithPrevMessage(
+                          node,
+                          (index - 1 >= 0) ? allNodes[index - 1].node : null,
+                          (index + 1 < allNodes.length) ? allNodes[index + 1].node : null,
```

Comment:
Please move this logic into the `shouldGroupWithPrevMessage` function

Reply:
I fixed this. Please check it.